server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileProcessor.java

```diff
@@ -576,6 +579,8 @@ public String getStorageGroupName() {
       QueryUtils.modifyChunkMetaData(chunkMetaDataList,
           modifications);
 
+      chunkMetaDataList.removeIf(chunkMetaData -> chunkMetaData.getEndTime() < timeLowerBound);
```

Comment:
Still the same question. The variable `timeLowerBound` will cause class construction every time the function is invoked.

Reply:
This will require maintaining a ThreadLocal predicate for each query. As far as I am concerned, this works no different from using a lambda since new object construction is unavoidable.

Somehow, I moved `timeLowerBound` into `QueryContext` and replaced this with a method reference.